### PR TITLE
feat(px): add peer-exchange v2alpha1 wire format

### PR DESCRIPTION
**Pull request by alrevuelta**
* Add peer-exchange v2alpha1 wire format
* See also https://github.com/vacp2p/rfc/pull/570

### CODE REVIEW

**Review by LNSD — APPROVED** [review]
LGTM ✅

**Comment by LNSD** [comment]
Added @danisharora099 (from js-waku) and @richard-ramos (from go-waku) as reviewers. We need confirmation that all implementations are using the same protobuf file

File: waku/peer_exchange/v2alpha1/peer_exchange.proto [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// 34/WAKU2-PEER-EXCHANGE rfc: https://rfc.vac.dev/spec/34/
+// Protocol identifier: /vac/waku/peer-exchange/2.0.0-alpha1
+package waku.peer_exchange.v2alpha1;
+
+message PeerInfo {
+  bytes enr = 1;
+}
+
+message PeerExchangeQuery {
+  uint64 num_peers = 1;
```
Comment (danisharora099) [inline waku/peer_exchange/v2alpha1/peer_exchange.proto:12]: `js-waku` uses camel case `numPeers`
would you suggest changing it?

File: waku/peer_exchange/v2alpha1/peer_exchange.proto [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+message PeerExchangeResponse {
+  repeated PeerInfo peer_infos = 1;
```
Comment (danisharora099) [inline waku/peer_exchange/v2alpha1/peer_exchange.proto:16]: camel case here as well

File: waku/peer_exchange/v2alpha1/peer_exchange.proto (hunk 1 above)
Comment (alrevuelta) [reply waku/peer_exchange/v2alpha1/peer_exchange.proto:12]: mm I think there is no need to change it. nwaku also uses camelCase.

I just changed it to comply with [protobuf](https://developers.google.com/protocol-buffers/docs/style#message_and_field_names) conventions, which is also enforced by the linter this repo has.

Its also quite common I would say to have this "discreppancy" snake vs camel. For example from:
https://developers.google.com/protocol-buffers/docs/reference/go-generated#fields

> Thus, the proto field foo_bar_baz becomes FooBarBaz in Go, and _my_field_name_2 becomes XMyFieldName_2.

**Review by richard-ramos — APPROVED** [review]

**Review by danisharora099 — APPROVED** [review]

File: waku/peer_exchange/v2alpha1/peer_exchange.proto (hunk 1 above)
Comment (jm-clius) [reply waku/peer_exchange/v2alpha1/peer_exchange.proto:12]: Convention is for protobuf to use `snake_case` for fields and implementations of that protobuf to use language rules (i.e. `camelCase` for fields in most cases).

**Review by jm-clius — APPROVED** [review]

**Comment by alrevuelta** [comment]
Does anyone know why I cant merge? cc @LNSD 

> Merging is blocked
The base branch does not allow updates. [Learn more about protected branches.](https://docs.github.com/articles/about-protected-branches/)

**Pull request merged**